ledger/ledger-common/src/main/kotlin/net/corda/v5/ledger/common/transaction/TransactionMetadata.kt

```diff
@@ -0,0 +1,59 @@
+package net.corda.v5.ledger.common.transaction
+
+/**
+ * This interface represents the metadata governing a transaction, capturing a snapshot
+ * of the system and software versions as the transaction was created.
+ */
+interface TransactionMetadata {
+
+    /**
+     * Returns the ledger model this transaction belongs to.
+     *
+     * @return the class name of the transaction implementation
+     */
+    fun getLedgerModel(): String
+
+    /**
+     * Returns the version of the ledger this transaction was created with
+     *
+     * @return the ledger version at creation time
+     */
+    fun getLedgerVersion(): Int
+
+    /**
+     * Returns the transaction subtype. This is ledger specific, check with the documentation of the
+     * ledger model you are using.
+     *
+     * @return Transaction subtype as string defined in the ledger model
+     */
+    fun getTransactionSubtype(): String?
+
+    /**
+     * Get information about the CPI running on the virtual node creating the transaction
+     *
+     * @return a summary of the CPI
+     */
+    fun getCpiMetadata(): CordaPackageSummary?
```

Comment:
If we're making `TransactionMetadata` public we should probably do something about the nullability of the metadata fields. Some of these are `?` and others aren't. I think they should either all be nullable or none be nullable and throw an error if its missing.

Reply:
AFAIK CPI should never be empty. (Although we only fill it with some fake data temporarily.)
`TransactionSubtype` is not really needed for Consensual ledger, but I think we agreed that we was going to put there a placeholder value ("normal"(?)), just to avoid diverging the two ledgers's metadata schemas just for this field.